src/mon/OSDMonitor.h

```diff
@@ -81,7 +81,7 @@ struct failure_info_t {
 			     MonOpRequestRef op) {
     map<int, failure_reporter_t>::iterator p = reporters.find(who);
     if (p == reporters.end()) {
-      if (max_failed_since < failed_since)
+      if (max_failed_since != utime_t() && max_failed_since < failed_since)
```

Comment:
I see how the second hunk is needed but this looks like it breaks the invariant that max_failed_since is the max over reporters when the first reporter is added...

Reply:
@liewegas 
We shall consider the following case:

- we have 3 reporters: [A, B, C]
- B is cancelled, __max_failed_since__ is invalidated(== utime_t())
- we get a new reporter D, without this change __max_failed_since__ will now unconditionally points to D
- we call __get_failed_since()__ to get __max_failed_since__, it won't be recalculated since __max_failed_since__ now points to D (not equal to utime_t())

Reply:
aha, got it!